sdk/rust/oak/src/storage/mod.rs

```diff
@@ -45,104 +43,30 @@ impl Storage {
-        // https://github.com/stepancheg/rust-protobuf/issues/455
-        request_any.set_type_url(
-            String::from("type.googleapis.com/") + operation_request.descriptor().full_name(),
-        );
-        operation_request
-            .write_to_writer(&mut request_any.value)
-            .expect("could not convert operation request to `Any` value");
-        let mut grpc_request = GrpcRequest::new();
-        grpc_request.method_name = grpc_method_name.to_owned();
-        grpc_request.set_req_msg(request_any);
-        let mut grpc_data = Vec::new();
-        grpc_request
-            .write_to_writer(&mut grpc_data)
-            .expect("could not convert gRPC request to bytes");
-
-        // Create an ephemeral channel for the response.
-        let (rsp_out, rsp_in) = match crate::channel_create() {
-            Ok(x) => x,
-            Err(status) => {
-                return Err(grpc::build_status(
-                    grpc::Code::INTERNAL,
-                    &format!("failed to create storage response channel: {:?}", status),
-                ));
-            }
-        };
-
-        crate::channel_write(self.write_channel, &grpc_data, &[rsp_out.handle])
-            .expect("could not write to channel");
-        crate::channel_close(rsp_out.handle).expect("could not close channel");
-
-        // Block until there is a response available.
-        loop {
-            let wait_result = crate::wait_on_channels(&[rsp_in]).unwrap();
-            if wait_result[0] == crate::ChannelReadStatus::ReadReady {
-                break;
-            }
-        }
-
-        let mut buffer = Vec::<u8>::with_capacity(256);
-        let mut handles = Vec::<crate::Handle>::with_capacity(1);
-        crate::channel_read(rsp_in, &mut buffer, &mut handles)
-            .expect("could not read from channel");
-        if !handles.is_empty() {
-            panic!("unexpected handles received alongside storage request")
-        }
-        let mut grpc_response: GrpcResponse = protobuf::parse_from_reader(&mut &buffer[..])
-            .expect("could not parse gRPC response from bytes");
-        info!(
-            "StorageChannelResponse: {}",
-            protobuf::text_format::print_to_string(&grpc_response)
+            protobuf::text_format::print_to_string(&operation_request)
         );
-        crate::channel_close(rsp_in.handle).expect("could not close channel");
-
-        let status = grpc_response.take_status();
-        if status.code != 0 {
-            Err(status)
-        } else {
-            let response = protobuf::parse_from_bytes(grpc_response.get_rsp_msg().value.as_slice())
-                .expect("could not parse gRPC response message from `Any` value");
-            Ok(response)
-        }
+        let type_url =
+            String::from("type.googleapis.com/") + operation_request.descriptor().full_name();
+        crate::grpc::invoke_grpc_method(
```

Comment:
(pre-existing) this method feels like should be defined on the gRPC client itself, WDYT?